src/components/AddressDetailExplorer.js

```diff
@@ -0,0 +1,528 @@
+      }
+      this.setState({ metadataLoaded: true });
+    });
+  }
+
+  /**
+   * Callback to be executed when user changes token on select input
+   *
+   * @param {String} Value of the selected item
+   */
+  onTokenSelectChanged = (value) => {
+    this.setState({
+      selectedToken: value,
+      metadataLoaded: false,
+      selectedTokenMetadata: null,
+      balance: {},
+      transactions: [],
+    }, () => {
+      this.updateTokenURL(value);
+      this.reloadTokenData(value);
+    });
+  }
+
+  /**
+   * Update URL with new selected token and trigger didUpdate
+   *
+   * @param {String} New token selected
+   */
+  updateTokenURL = (token) => {
+    const newURL = this.pagination.setURLParameters({ token });
+    this.props.history.push(newURL);
+  }
+
+  /**
+   * Check if the searched address is on the inputs or outputs of the new tx
+   *
+   * @param {Object} tx Transaction data received in the websocket
+   * @param {boolean} checkToken If should also check if token is the same, or just address
+   *
+   * @return {boolean} True if should update the list, false otherwise
+   */
+  shouldUpdate = (tx, checkToken) => {
+    const arr = [...tx.outputs, ...tx.inputs];
+    const token = this.pagination.obtainQueryParams().token;
+
+    for (const element of arr) {
+      if (element.decoded.address === this.state.address) {
+        // Address is the same
+        if ((checkToken && element.token === token) || !checkToken) {
+          // Need to check token and token is the same, or no need to check token
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Redirects to transaction detail screen after clicking on a table row
+   *
+   * @param {String} hash Hash of tx clicked
+   */
+  onRowClicked = (hash) => {
+    this.props.history.push(`/transaction/${hash}`);
+  }
+
+  /**
+   * Refresh web page
+   *
+   * @param {Event} e Click event
+   */
+  refreshPage = (e) => {
+    e.preventDefault();
+    window.location.reload();
+  }
+
+  /**
+   * Refresh all data for the selected token
+   *
+   * @param {Event} e Click event
+   */
+  refreshTokenData = (e) => {
+    e.preventDefault();
+    this.setState({ showReloadTokenButton: false }, () => {
+     this.reloadTokenData();
+    })
+  }
+
+  /**
+   * Refresh all data.
+   *
+   * @param {Event} e Click event
+   */
+  refreshPageData = (e) => {
+    e.preventDefault();
+    this.setState({ showReloadDataButton: false }, () => {
+     this.reloadData();
+    })
+  }
+
+  numPages = () => {
+    return Math.ceil(this.state.balance.transactions / TX_COUNT);
+  }
+
+  render() {
+    const renderWarningAlert = () => {
+      if (this.state.warningRefreshPage) {
+        return (
+          <div className="alert alert-warning refresh-alert" role="alert">
+            There is a new transaction for this address. Please <a href="true" onClick={this.refreshPage}>refresh</a> the page to see the newest data.
```

Comment:
Maybe we could use `refreshPageData` here, what do you think?

Reply:
Done

Reply:
Have you tested this? I'm afraid this won't work if you are not in page 1.

You will probably load the same page you are because you will get the page from state, or not?

Reply:
Created a method to clear the query parameters then call `refreshPageData` this will make all parameters go to default except the address and selected token, i.e. it will go to the first page of the current token.